pkg/util/logging/persist_logger.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type PersistLogger interface {
+	logrus.FieldLogger
+	PersistLogStop(log logrus.FieldLogger)
+	GetPersistLogFile() (*os.File, error)
+	Dispose(log logrus.FieldLogger)
+}
+
+type persistLogger struct {
+	logrus.FieldLogger
+	logger *logrus.Logger
+	file   *os.File
+	w      *gzip.Writer
+	fields logrus.Fields
+}
+
+// NewPersistLogger creates the logger instance that writes logs to both Stdout and a file in the temp folder so that
+// it can be persisted later
+func NewPersistLogger(logLevel logrus.Level, logFormat Format, hook *LogHook, fields logrus.Fields) (PersistLogger, error) {
```

Comment:
Trvial:
IMO we better call this a NewTempFileLogger, b/c it writes the content to a temp file.

Reply:
I changed the name to DualModeLogger and changed the function names accordingly.
If we name it as TempFileLogger, the call is somehow confusing like below:
`tempFileLogger.TempFileLogStop`

It is confusing that once `tempFileLogger.TempFileLogStop` is called, what else it does (when we call logger.Info again) since the name is tempFileLogger and the TempFileLog has been stopped. Actually, it is still functioning and write the log to stdout only.

Reply:
As the discussion, changed the interface to DueModeLogger and changed the method names accordingly; changed the instance name to tempFileLogger